airbyte-cdk/python/airbyte_cdk/v2/abstract_source_v2.py

```diff
@@ -0,0 +1,65 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from typing import Mapping, Any, TypeVar, Iterable, Optional, Union, List, Tuple
+
+from airbyte_cdk.sources import Source
+
+from airbyte_cdk.sources.streams.core import StreamData
+from airbyte_protocol.models import ConfiguredAirbyteCatalog, AirbyteCatalog, AirbyteConnectionStatus, Status
+
+from airbyte_cdk.v2 import Stream
+
+
+class AbstractSource(Source, ABC):
```

Comment:
I don't see where this class is used. Is it cruft / replaced by AsyncSource?

Reply:
nvm. I see that the demo does not run a full source. The main follow up item I would be interested in is whether there's plausible adapter that would allow us to upgrade AbstractSource and HttpStream to use this v2 without changing their interface